external_executors/local_cromwell/local_cromwell.py

```diff
@@ -103,16 +103,47 @@
 # that refer to remote locations with entries that refer to the local paths.
 # TODO: Handle protocols other than http (gs, s3, ...)
 python3 << END
-import json, os, subprocess
+import json, os, subprocess, tempfile
+from google.cloud import storage
 inputs_dict = json.load(open("inputs.json"))
-localized_inputs_dict = dict()
-for entry, value in inputs_dict.items():
-    if value.startswith("https://") or value.startswith("http://"):
-        local_path = os.path.basename(value)
-        subprocess.run(["curl", "-o", local_path, value], check=True)
-        localized_inputs_dict[entry] = local_path
+
+def match_url(s):
+    return s.startswith("http://") or \
+           s.startswith("https://") or \
+           s.startswith("gs://")
+
+def localize_url(url):
+    tmp_dir = tempfile.mkdtemp()
+    if url.startswith("https://") or url.startswith("http://"):
+        local_path = os.path.join(tmp_dir, os.path.basename(url))
+        subprocess.run(["curl", "-o", local_path, url], check=True)
+        return local_path
+    elif url.startswith("gs://"):
+        client = storage.client.Client.create_anonymous_client()
+        path_parts = url[5:].split("/")
+        bucket_name = path_parts[0]
+        bucket = client.bucket(bucket_name)
+        blob = bucket.blob('/'.join(path_parts[1:]))
+        local_path = os.path.join(tmp_dir, path_parts[-1])
+        blob.download_to_filename(local_path)
+        return local_path
+    else:
+        return url
+
+def localize_inputs(obj, match_func, localize_func):
+    if isinstance(obj, list):
```

Comment:
would make this `isinstance(obj, (list, tuple))`

Reply:
I can change this, though I think we're only going to get types produces by `json.load`, which shouldn't make a `tuple`: https://github.com/python/cpython/blob/3.6/Lib/json/decoder.py#L258-L276